Orange/classification/tree.py

```diff
@@ -1,18 +1,191 @@
+            return best_score, node, branches, 2
+
+        #######################################
+        # The real _select_attr starts here
+        domain = data.domain
+        class_var = domain.class_var
+        best_score, *best_res = REJECT_ATTRIBUTE
+        best_res = [Node(None, None, None)] + best_res[1:]
+        disc_scorer = _score_disc_bin if self.binarize else _score_disc
+        for attr_no, attr in enumerate(domain.attributes):
+            sc, *res = disc_scorer() if attr.is_discrete else _score_cont()
+            if res[0] is not None and sc > best_score:
+                best_score, best_res = sc, res
+        best_res[0].value = distribution.Discrete(data, class_var)
+        return best_res
+
+    def build_tree(self, data, active_inst, level=1):
+        """Induce a tree from the given data
+
+        Returns:
+            root node (Node)"""
+        node_insts = data[active_inst]
+        distr = distribution.Discrete(node_insts, data.domain.class_var)
+        if len(node_insts) < self.min_samples_leaf:
+            return None
+        if len(node_insts) < self.min_samples_split or \
+                max(distr) >= sum(distr) * self.sufficient_majority or \
+                self.max_depth is not None and level > self.max_depth:
+            node, branches, n_children = Node(None, None, distr), None, 0
+        else:
+            node, branches, n_children = self._select_attr(node_insts)
+        node.subset = active_inst
+        if branches is not None:
+            node.children = [
+                self.build_tree(data, active_inst[branches == br], level + 1)
+                for br in range(n_children)]
+        return node
+
+    def fit_storage(self, data):
+        if self.binarize and any(
+                attr.is_discrete and len(attr.values) > self.MAX_BINARIZATION
+                for attr in data.domain.attributes):
+            # No fallback in the script; widgets can prevent this error
+            # by providing a fallback and issue a warning about doing so
+            raise ValueError("Exhaustive binarization does not handle "
+                             "attributes with more than {} values".
+                             format(self.MAX_BINARIZATION))
+
+        active_inst = np.nonzero(~np.isnan(data.Y))[0]
+        root = self.build_tree(data, active_inst)
+        if root is None:
+            distr = distribution.Discrete(data, data.domain.class_var)
+            if np.sum(distr) == 0:
+                distr[:] = 1
+            root = Node(None, 0, distr)
+        root.subset = active_inst
+        model = OrangeTreeModel(data, root)
+        return model
+
+
+class TreeClassifier(SklModel):
+    """Wrapper for SKL's tree classifier with the interface API for
+    visualizations"""
+    def __init__(self, *args, **kwargs):
+        SklModel.__init__(self, *args, **kwargs)
+        self._cached_sample_assignments = None
 
 
 class TreeLearner(SklLearner):
```

Comment:
Why not make our tree learner the default one and rename this one to `SklTreeLearner`?